tests/unit/test_repos_branch_protection.py

```diff
@@ -0,0 +1,207 @@
+"""Unit tests for methods implemented on Branch Protection."""
+import github3
+from github3.repos.branch import BranchProtection
+from . import helper
+from json import dumps
+import pytest
+
+protection_example_data = helper.create_example_data_helper(
+    'branch_protection_example'
+)
+protection_url_for = helper.create_url_helper(
+    'https://api.github.com/repos/octocat/Hello-World/'
+    'branches/master/protection'
+)
+
+
+# class TestBranchProtection(helper.UnitRequiresAuthenticationHelper):
+#     """Branch protection unit test"""
+#
+#     described_class = github3.repos.branch.BranchProtection
+#     example_data = protection_example_data()
+#
+#     def test_protect(self):
+#         """Verify PUT method works"""
+#         self.instance.enforce_admins = True
+#         self.instance.restrictions = ['continuous-integration/travis-ci']
+#         self.instance.update()
+#         self.put_called_with(url_for())
+#
+#     def test_unprotect(self):
+#         """Verify DELETE method works"""
+#         self.instance.delete()
+#         self.delete_called_with(url_for())
+
+
+enforce_admins_url_for = helper.create_url_helper(
+    'https://api.github.com/repos/octocat/Hello-World/'
+    'branches/master/protection/enforce_admins'
+)
+enforce_admins_example_data = helper.create_example_data_helper(
+    'branch_protection_enforce_admins_example')
+
+
+class TestProtectionEnforceAdmins(helper.UnitHelper):
+    described_class = github3.repos.branch.ProtectionEnforceAdmins
+    example_data = enforce_admins_example_data()
+
+    def test_enable(self):
+        self.instance.enable()
+        self.post_called_with(enforce_admins_url_for(),
+                              headers=BranchProtection.PREVIEW_HEADERS_MAP)
+
+    def test_disable(self):
+        self.instance.disable()
+        self.delete_called_with(enforce_admins_url_for(),
+                                headers=BranchProtection.PREVIEW_HEADERS_MAP)
+
+    def test_update_delete(self):
+        self.instance.enabled = False
+        self.instance.update()
+        self.delete_called_with(enforce_admins_url_for(),
+                                headers=BranchProtection.PREVIEW_HEADERS_MAP)
+
+    def test_update_post(self):
```

Comment:
As I feared, this is not how `update` should work. Please remove that method and these tests.